paasta_tools/frameworks/task_store.py

```diff
@@ -0,0 +1,206 @@
+import json
+
+from kazoo.client import KazooClient
+from kazoo.exceptions import BadVersionError
+from kazoo.exceptions import NodeExistsError
+from kazoo.exceptions import NoNodeError
+
+from paasta_tools.utils import _log
+
+
+class MesosTaskParametersIsImmutableError(Exception):
+    pass
+
+
+class MesosTaskParameters(object):
+    def __init__(
+        self,
+        health=None,
+        mesos_task_state=None,
+        is_draining=None,
+        is_healthy=None,
+        offer=None,
+        resources=None,
+    ):
+        self.__dict__['health'] = health
+        self.__dict__['mesos_task_state'] = mesos_task_state
+        self.__dict__['is_draining'] = is_draining
+        self.__dict__['is_healthy'] = is_healthy
+        self.__dict__['offer'] = offer
+        self.__dict__['resources'] = resources
```

Comment:
sorry to grave dig an old commit, but I thought you should know :)

assigning to `self.__dict__['x'] = ...` is better written as `self.x = ...`

I initially thought this was an attempt to side-step mypy, but mypy wasn't in play at this point...

Reply:
oh I see the `__setattr__` / `__delattr__` below, tricky tricky